### DISCUSSION TITLE
Laser not working (Arduino Nano board)

### DISCUSSION THREAD

**Discussion by bssssy** [arkypita/LaserGRBL#1409] [Q&A]
Hi everybody, I need some help with my laser :(....

I just bought my laser engraver, which cames with some chinese program ("Lite Fire Laser"). Although I made some tests with the laser engraver, said program was very poor, and give me some "bad feelings", in this manner I look for other programs and found Lasergrbl (which is everything i need!). 
However, after flashing the Arduino Nano and follow the instructions, i cant manage to make the laser works (I can only move the motors).

I've tried with $32=0 and $32=1; without good results.

I've also follow the "how guide #1232" (since the board is similar), without results.

Herebelow is a picture of the board, i'm not sure if it is compatible with lasergrbl (however, the chinese program uses grbl).

Any idea of what is happening?  

lot and tons of thanks, in advance  :)
![placa](https://user-images.githubusercontent.com/80602172/114481041-b9a4fd80-9bd1-11eb-9167-3b203c990f64.jpg)

**Comment by arkypita** [comment]
Does the laser work with the original software or not?

**Reply by bssssy** [reply]
hi, it used to work. After flashing the arduino the laser and the motors have stop working with the original software.

**Reply by arkypita** [reply]
> I just bought my laser engraver, which cames with some chinese program ("Lite Fire Laser"

Do you have a link where to download the bundled "Lite Fire Laser" software?

> I've also follow the "how guide #1232" (since the board is similar), without results.

This link refer to a totally different issue.

> Herebelow is a picture of the board, i'm not sure if it is compatible with lasergrbl (however, the chinese program uses grbl).
> Any idea of what is happening?

Are you sure of that? What version of grbl was originally installed on controller board? 0.8? 0.9? 1.1?

**Reply by arkypita** [reply]
More info about your laser engraver? What model is it? Picture of laser module?

I see your laser has 2 wire, usually there are 3 wire going to laser module: GND, VCC (+12v) and PWM (+5v TTL power modulation).

**Reply by Stevyjb** [reply]
Hi, More Information is required ! Is lasergrbl actually connecting to the controller Board?

**Reply by bssssy** [reply]
> > I just bought my laser engraver, which cames with some chinese program ("Lite Fire Laser"
> 
> Do you have a link where to download the bundled "Lite Fire Laser" software?
> 
> > I've also follow the "how guide #1232" (since the board is similar), without results.
> 
> This link refer to a totally different issue.
> 
> > Herebelow is a picture of the board, i'm not sure if it is compatible with lasergrbl (however, the chinese program uses grbl).
> > Any idea of what is happening?
> 
> Are you sure of that? What version of grbl was originally installed on controller board? 0.8? 0.9? 1.1?

Thanks for the response, here is a link with the software https://we.tl/t-ziFyLmTpHR 

I also dont know the version of grbl originally installed on the board (sry for the lack of information)

here are some pictures:

![4](https://user-images.githubusercontent.com/80602172/114643260-571c3200-9ca3-11eb-97ca-26d35154a017.jpg)
![1](https://user-images.githubusercontent.com/80602172/114643262-584d5f00-9ca3-11eb-9116-7904e7c03a44.jpg)
![2](https://user-images.githubusercontent.com/80602172/114643263-584d5f00-9ca3-11eb-9029-a03a6b18c18d.jpg)
![3](https://user-images.githubusercontent.com/80602172/114643265-58e5f580-9ca3-11eb-8751-d0dfbcdc3e20.jpg)

**Reply by bssssy** [reply]
> More info about your laser engraver? What model is it? Picture of laser module?
> 
> I see your laser has 2 wire, usually there are 3 wire going to laser module: GND, VCC (+12v) and PWM (+5v TTL power modulation).

I bought the laser engraver from ebay to "lx diy alliance", I dont have more information (im searching in old emails)

regarding the 2 wire, I assume that the laser would not support the pwm (I set $32=0)

**Reply by bssssy** [reply]
> Hi, More Information is required ! Is lasergrbl actually connecting to the controller Board?

Hi, yep, it can connect, and I can also move the stepper motors.

(thanks)

**Reply by Stevyjb** [reply]
Hello, Will the engraver run the job/program ( It makes all the movements but just laser not on ). If this is the case it seems  Lasergrbl is connected and working correctly. Check the laser  switch and its power settings and its connections to to the board. It may be a hardware problem

**Reply by bssssy** [reply]
I think the problem is related (as arkpita mention) with the pin connection of the board version :c

**Comment by arkypita** [comment]
Your machine look like a benbox controller, so not grbl. https://www.youtube.com/watch?v=391LFYf9DTc
Maybe can work with grbl 1.1 too, but can be necessary pin remapping.

Older GRBL 0.8 and 0.9 (and maybe benbox) use pin D12 for spindle enable (or laser ON/OFF)
https://github.com/grbl/grbl/wiki/Connecting-Grbl

Grbl 1.1 use pin D11 for spindle PWM 
https://github.com/gnea/grbl/wiki/Connecting-Grbl

Use a multimeter to try understand what is the pin that control your laser.

![image](https://user-images.githubusercontent.com/8782035/114933439-f3c40880-9e38-11eb-83e6-67df097e7cd7.png)

**Reply by bssssy** [reply]
Thanks a lot, im gonna finde which pin that control the laser.

**Comment by arkypita** [comment]
Well... I did some test with litefire.
Your engraver use grbl 0.8 with 9600 baud

Here is a link of the pre-complied hex file: [Grbl v0.8c Atmega328p 16mhz 9600baud](http://bit.ly/SSdCJE)

**Reply by arkypita** [reply]
this is why:

1) after flashing the original software no longer connect (v1.1 use 115200 litefire use 9600 hardcoded)
2) with lasergrbl and other sw laser does not turn on (different pin mapping)

**Reply by bssssy** [reply]
thanks again (truly), in view of the situation and your explication I'm thinking of buying a new board and a laser which allows pwm (and leave this board for a future project). 
do you recommend a specific board compatible with lasergrbl? (I really like the features of the lasergrbl software)

thx again!

**Reply by arkypita** [reply]
I have no direct experience with other board than CNC shield V3 (which I do not recommend because it is without connectors). Try asking to lasergrbl community https://www.facebook.com/groups/lasergrbl.

I believe you can use the board you have with grbl 0.9 and variable spindle PWM DISABLED

**Reply by bssssy** [reply]
Thanks! im gonna try :)

**Reply by Stevyjb** [reply]
https://s.click.aliexpress.com/e/_AUY8Mj

This is a grbl controller board for duel "Y" Stepper motors

**Reply by bssssy** [reply]
> I have no direct experience with other board than CNC shield V3 (which I do not recommend because it is without connectors). Try asking to lasergrbl community https://www.facebook.com/groups/lasergrbl.
> 
> I believe you can use the board you have with grbl 0.9 and variable spindle PWM DISABLED

hey, thx! I tried with 0.9 version and it works fine :) (i've load the hex file from the grbl page and upload it with xload), in addition i've disable the pwm option, now is functioning without problems.

Thanks for the help!

anyways, i'm thinking of buying a new laser and board, in order to enable the modulation.
:)

**Comment by arkypita** [comment]
Good to know!
